doc/development.md

```diff
@@ -89,4 +89,205 @@ merged, we need to test them on all supported distributions and
 architectures.
 
 When you push patches to GitHub, CI will run its tests according to the
-configuration in the `.github/workflows/ci.yml` file.
\ No newline at end of file
+configuration in the `.github/workflows/ci.yml` file.
+
+## Coding guidelines
+
+The coding conventions in Vdsm are generally aligned with the
+[PEP 8](https://peps.python.org/pep-0008/) standard guidelines.
+Code is linted with `pylint` based in the `/.pylintrc` file.
+You can lint the code before submitting a patch by doing:
+
+    make lint
+
+However, some code conventions are not explicitly enforced by the linter due
+to legacy code and inconsistent styles. But new code shall adhere to those
+rules, and old code, should be updated as possible when patched.
+
+In this section, we will cover some of the most important conventions and
+preferences for contributing to Vdsm.
+
+### Imports
+
+First import standard library modules, then 3rd party modules, and finally
+project modules.
+Separate standard library modules and 3rd party modules with a blank line.
+Keep alphabetical order within each block.
+
+Less specific imports:
+
+    from vdsm.foo import baz
+
+Should come before more specific imports like:
+
+    from vdsm.foo.bar import baz
+
+Accepted, but better to avoid:
+- Relative name imports
+- Import many modules in a comma separated list
+- Alias an imported module with non-standard abbreviation
+
+### Indentation
+
+Use 4 spaces per indentation level.
+
+Continuation lines should align wrapped elements either vertically using
+Python's implicit line joining inside parentheses, brackets and braces, or
+using a hanging indent. When using a hanging indent, the following should
+be considered:
+- There should be no arguments on the first line.
+- Further indentation should be used to clearly distinguish itself as a continuation line.
+- Hanging indentation should stick to the 4-space rule.
+```python
```

Comment:
This is not clear. It would be useful to discus special cases with examples:
- How to define functions with many arguments
- How to call function with many arguments
- How to call function with positional and keyword arguments
- When to use conditional expressions (a if condition else b)
- How to format list comprehension

Reply:
I think many of these points are already covered in the text and examples.